src/App.tsx

```diff
@@ -1,14 +1,48 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api/api';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoId, setTodoId] = useState(0);
+  const [loading, setLoading] = useState(true);
+  const [selectedFilter, setSelectedFilter] = useState(Status.All);
+  const [query, setQuery] = useState('');
+
+  const loadTodos = async () => {
+    const loadedTodos = await getTodos();
+
+    setTodos(loadedTodos);
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  const searchedTodos = todos.filter((todo) => todo
+    .title.trim().toLowerCase().includes(query.trim().toLowerCase()));
+
+  const filteredTodos = useMemo(() => {
```

Comment:
useMemo is redundant here, 'cause searchedTodos dependency re-creating on every rerender